tests/plugin_filter/util.py

```diff
@@ -0,0 +1,218 @@
+# Copyright 2022 University of Stuttgart
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from qhana_plugin_registry.db.models.plugins import RAMP, PluginTag
+from qhana_plugin_registry.db.models.templates import UiTemplate, TemplateTab
+from qhana_plugin_registry.tasks.plugin_filter import apply_filter_for_tab
+
+import random
+from hypothesis import strategies as st
+from packaging.specifiers import Specifier
+import json
+from sqlalchemy.sql import exists
+
+
+def filter_strategy():
+    return st.one_of(
+        st.fixed_dictionaries({"tag": st.text()}),
```

Comment:
Currently, the empty string for names/tags is handled just like every other name/tag. Hypothesis creates plugins with empty names and an "empty tag", which match the filters `{"name": ""}` and `{"tag": ""}` respectively.

Reply:
The database may not prevent this, but I sure hope we prevent empty names in the API models. If we do, this test can assume non-empty names. If we don't, please open an issue so that we do in the future...

Reply:
The empty string as plugin name is not allowed. However, the empty string as tag is allowed, I.e. if the plugin runner provides a plugin that has the empty string as one of its tags, the QHAna UI shows an empty tag.

I'll assume plugin names of length `>= 1` in the tests.

Reply:
The empty string as plugin name is not allowed. However, the empty string as tag is allowed, I.e. if the plugin runner provides a plugin that has the empty string as one of its tags, the QHAna UI shows an empty tag.

I'll assume plugin names of length `>= 1` in the tests.

Reply:
You can assume the same thing for tags. I guess we can just ignore an empty tag in plugins without throwing an error (but we should not add empty tags to the database). Can you create an issue for that (that empty tags get added to the DB)?

Reply:
I set `min_size=1` for tag names and created an issue (#14).